client/src/request.js

```diff
@@ -1,38 +1,63 @@
-const photos = (productId) => {
-  return fetch(`http://ec2-3-136-203-39.us-east-2.compute.amazonaws.com:4002/photos/id/${productId}`)
-  .then(res => res.json())
-  .then(productPhotos => {
-    return productPhotos;
-  })
-  .catch(error => {
+const fetchWithTimeout = async (resource, options) => {
```

Comment:
Can you clarify for me what is going on with this code between lines 1 - 14.

Reply:
I got this function from [this site](https://dmitripavlutin.com/timeout-fetch-request/)